java_src/foam/nanos/auth/AuthOAO.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.auth;
+
+import foam.core.FObject;
+import foam.core.X;
+import foam.oao.ProxyOAO;
+
+import javax.security.auth.AuthPermission;
+import java.util.Map;
+
+public class AuthOAO
+  extends ProxyOAO
+{
+  protected String rootPermission_ = null;
+
+  public AuthOAO(String rootPermission_) {
```

Comment:
Name the argument 'rootPermission' without the underscore. Just the instance variable has the _.